plugins/aws/openSearchServerless/collectionPublicAccess.js

```diff
@@ -0,0 +1,87 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Collection Public Access',
+    category: 'OpenSearch',
+    domain: 'Serverless',
+    description: 'Ensures that OpenSearch Serverless collections are not publicly accessible',
+    more_info: 'OpenSearch Serverless collections should be not be publicly accessible to prevent unauthorized actions.',
+    link: 'https://docs.aws.amazon.com/opensearch-service/latest/developerguide/serverless-network.html',
+    recommended_action: 'Update the network policy and remove the public access to collection.',
+    apis: ['OpenSearchServerless:listNetworkSecurityPolicies', 'OpenSearchServerless:getNetworkSecurityPolicy', 'OpenSearchServerless:listCollections'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        async.each(regions.opensearchserverless, function(region, rcb){
+            var listCollections = helpers.addSource(cache, source, 
+                ['opensearchserverless', 'listCollections', region]);
+            
+            if (!listCollections) return rcb();
+
+            if ( !listCollections.data || listCollections.err) {
+                helpers.addResult(results, 3,
+                    'Unable to query list collections: ' + helpers.addError(listCollections), region);
+                return rcb();
+            }
+            if (!listCollections.data.length){
+                helpers.addResult(results, 0, 'No Collection found', region);
+                return rcb();
+            }
+            var listSecurityPolicies = helpers.addSource(cache, source,
+                ['opensearchserverless', 'listNetworkSecurityPolicies', region]);
+
+            if (!listSecurityPolicies && listSecurityPolicies.err || !listSecurityPolicies.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query list security policy: ' + helpers.addError(listSecurityPolicies), region);
+                return rcb();
+            }
+
+            if (!listSecurityPolicies.data.length) {
+                helpers.addResult(results, 0, 'No Security Policy found', region);
```

Comment:
we should rcb if no security policies found. no policies means collection does not have any policy attached so it is not publicly accessible